[FMS] Accessibility improvements

lucascumsille opened the pull request:
I tried splitting the commits so I can drop them as needed.

### Heading hierarchy
Fixes: https://github.com/mysociety/societyworks/issues/3428
As we spoke, the repeating H1 where not issue, because only only is visible.
Related commits:

**Around Page**
- [[Bromley] Around Page: H2 -> H1](https://github.com/mysociety/fixmystreet/pull/4312/commits/97ef2f1da6e23875a1135aa10c39fc4912bd13bc) 
- [[Greenwich] Around Page: H2 -> H1](https://github.com/mysociety/fixmystreet/pull/4312/commits/3ebf1901aefdcb8a8058695917bd2d1dfb27adb0)
- [[Oxfordshire] Around Page: H2 -> H1](https://github.com/mysociety/fixmystreet/pull/4312/commits/290832dd90cbbaae949762bffe27f28a929a59a1)
- [[FMS] Reduced H1 size for click-the-map element](https://github.com/mysociety/fixmystreet/pull/4312/commits/5324fc91277a53f698c0183b90718884e0e1776d)

Related, but more UI related. I got rid of the yellow chevron background on this commit:
[[FMS] Erased odd chevron bakground click-the-map element](https://github.com/mysociety/fixmystreet/pull/4312/commits/527e036525b056e9ab30111a613758c797a5c0be)
<img width="457" alt="Screenshot 2023-02-28 at 10 51 37" src="https://user-images.githubusercontent.com/13790153/222087047-4473ec47-1ca6-444a-8d65-8ab57d3cf467.png">

**Report page**

- [[FMS][Report Updates Heading] Replace h4.static-with-rule for h2.stat…](https://github.com/mysociety/fixmystreet/pull/4312/commits/3fd6b6ab96f26af698ffb216d5ce994298c7ecc1)
- [[Zurich][Report Updates Heading] Replace h4.static-with-rule for h2.s…](https://github.com/mysociety/fixmystreet/pull/4312/commits/720c687e44a17180ce231e4ec314823c0a0b99db)

### Greenwich Improvements to navbar.

The mobile navbar was overlapping the desktop one. Leaving a very small clickable area.
- [[Greenwich] Increase the clickable area for top navbar](https://github.com/mysociety/fixmystreet/pull/4312/commits/9119965618803b5717e2c4f5c159841c76c32008)

Close mobile icon was white, therefore invisible.
- [[Greenwich] Added close button to mobile navbar](https://github.com/mysociety/fixmystreet/pull/4312/commits/3119c22a7bbbd015d9114574f30272ddd45e92cc)


### Mixin button-variant
Fixes: https://github.com/mysociety/societyworks/issues/3333

This one separate the focus state with the hover effect. Because all cobrands will be using the default mixing yellow, we could have some cobrands complaining about their styleguide, but I still thinks this is more accessible of what we currently have.
[[FMS] Added focus state to button-variant mixin](https://github.com/mysociety/fixmystreet/pull/4312/commits/ff3c30ba465f326fa1432e229da5cd0c876aea4f)

This one adds an outline to the focus state in case we wanted to make it more obvious. But we can drop it if you think is too much.
[[FMS] Added Outline property to button-variant mixin](https://github.com/mysociety/fixmystreet/pull/4312/commits/802f965d85cf807fd368a30be08caac5b0811628)


### Focus state for local images
Fixes: https://github.com/mysociety/societyworks/issues/3298
<img width="1187" alt="Screenshot 2023-03-01 at 07 47 26" src="https://user-images.githubusercontent.com/13790153/222089893-33f37195-f242-4853-ad88-d2a7f33fc3f5.png">

[[FMS] Added focus state to local alert images page](https://github.com/mysociety/fixmystreet/pull/4312/commits/469afd2f79915596129109836b628cbc554ae0e0)

I hope it's easy to understand, but let me know if you need me to clarify anything.

dracos commented on templates/web/base/report/update.html:
```diff
@@ -1,7 +1,7 @@
 [% IF loop.first %]
 <section class="full-width">
-    <h4 class="static-with-rule">[% loc('Updates') %]</h4>
+    <h2 class="static-with-rule">[% loc('Updates') %]</h2>
```
The fixamingata and oxfordshire layout.scss also have a h4 that will need changing

dracos commented on web/cobrands/sass/_mixins.scss:
```diff
@@ -78,6 +78,7 @@ $button-margin-left: 0 !default;
     &:focus{
       color: $focus-text !important;
       background: $focus-bg-bottom;
+      outline: $focus-outline-width solid $focus-outline-color;
```
This doesn't use the $focus-outline-style passed in, should it be?

dracos replied:
Sorry, didn't answer your question about outline - I don't know, the focus colour does seem like it might be enough? I can't see outline used on e.g. gov.uk like this.

dracos commented on web/cobrands/sass/_mixins.scss:
```diff
@@ -12,6 +12,11 @@ $button-margin-bottom: 0 !default;
 $button-margin-right: 0 !default;
 $button-margin-left: 0 !default;
 
+// Focus Outline
```
These seem very generically named if they're only being used for the focus within `.alerts__nearby-activity__photos`. Are these likely to be overridden? I wouldn't think so, so I'm not sure it's worth having these as variables until such a time as that proves necessary.

dracos commented on CHANGELOG.md:
```diff
@@ -43,6 +43,9 @@
         - Remove unnecessary and add some missing fieldsets.
         - Improve fancybox accessibility (text and focus).
         - Add visually-hidden nearest address in report list.
+        - Add a focus state for button-variant mixin. Now Hover and Focus stare are different.
```
Typo of 'state', but I think second sentence can just be dropped

dracos commented on CHANGELOG.md:
```diff
@@ -43,6 +43,9 @@
         - Improve fancybox accessibility (text and focus).
         - Add visually-hidden nearest address in report list.
+        - Add a focus state for button-variant mixin. Now Hover and Focus stare are different.
+        - For cobrands that use the "click-the-map" element the heading hierarchy has been fixed, changing the h2 tag to h1.
```
This doesn't apply to the core, so doesn't need a changelog line.

dracos commented on web/cobrands/sass/_mixins.scss:
```diff
@@ -62,8 +67,7 @@ $button-margin-left: 0 !default;
 
-    &:hover,
-    &:focus {
+    &:hover{
```
Can we have a space before the `{`s to match everywhere else

dracos reviewed (changes requested):
All looks good - hopefully my comments all make sense, let me know if not!

lucascumsille commented:
Thanks for the feedback @dracos =). I implemented everything stated above. How you prefer me to rebase this? All the commits that are related into one? 

One extra thing I noticed was the Button for the Dropzone on Fixamingata. It wasn't accessible enough so I changed the color for the button to black:

<img width="461" alt="Screenshot 2023-03-02 at 06 58 45" src="https://user-images.githubusercontent.com/13790153/222358811-62badc48-dbf6-4b30-9ab4-c174a7c36180.png">

<img width="461" alt="Screenshot 2023-03-02 at 06 52 01" src="https://user-images.githubusercontent.com/13790153/222358803-a95b5257-6106-414e-b2e5-2809cbbd387f.png">

dracos commented on web/cobrands/sass/_base.scss:
```diff
@@ -258,18 +258,15 @@ a,
 }
 
-h4.static {
+h2.static, h2.static-with-rule {
```
If they're being combined, might as well drop the h2.static? The point of them being separate was static-with-rule added a border-top etc that the static class did not have - if they're now the same thing, and static is unused, might as well delete it.

dracos reviewed (changes requested):
One comment to remove `h2.static` is all, plus there's a map file committed by mistake.
Would be great if, in general, fixes on a PR could be committed as extra 'fixup' commits applying to the commits they are fixing, which makes it much easier to then re-review the changes - let me know if you'd like more information on that :)
In terms of squashing - yes, I would put all the updates heading in one commit, all the around page ones in one commit, the rest all make sense as separate commits, I think?

lucascumsille commented:
Thanks @dracos I'll do the rebase squashing related commits. I'll include a fixup getting rid of the `h4.static` that is not being used first. 
Thanks for adding your commit =)

lucascumsille commented:
@dracos I just pushed a --fixup getting read of the h2.static element. Let me know if you are happy with the changes and I'll rebase =)

dracos reviewed (approved):
The map file is still there needs removing as part of the rebase as well. Once you've tidied it up, let me have a final look before merging, thanks :)
